lib/membrane_h264_ffmpeg/encoder.ex

```diff
@@ -170,18 +174,22 @@ defmodule Membrane.H264.FFmpeg.Encoder do
   end
 
   defp flush_encoder_if_exists(%{encoder_ref: encoder_ref, use_shm?: use_shm?}) do
-    with {:ok, dts_list, frames} <- Native.flush(use_shm?, encoder_ref) do
-      buffers = wrap_frames(dts_list, frames)
+    with {:ok, dts_list, pts_list, frames} <- Native.flush(use_shm?, encoder_ref) do
+      buffers = wrap_frames(dts_list, pts_list, frames)
       {:ok, buffers}
     end
   end
 
-  defp wrap_frames([], []), do: []
+  defp wrap_frames([], [], []), do: []
 
-  defp wrap_frames(dts_list, frames) do
-    Enum.zip(dts_list, frames)
-    |> Enum.map(fn {dts, frame} ->
-      %Buffer{dts: Common.to_membrane_time_base_truncated(dts), payload: frame}
+  defp wrap_frames(dts_list, pts_list, frames) do
+    Enum.zip([dts_list, pts_list, frames])
+    |> Enum.map(fn {dts, pts, frame} ->
+      %Buffer{
+        pts: Common.to_membrane_time_base_truncated(pts),
+        dts: Common.to_membrane_time_base_truncated(dts),
```

Comment:
As discussed, there are problems with handling [`AV_NOPTS_VALUE`](https://ffmpeg.org/doxygen/trunk/group__lavu__time.html#ga2eaefe702f95f619ea6f2d08afa01be1) constant.
1. The decoder handles it at the input, but the conversion of `AV_NOPTS_VALUE` -> `nil` after decoding is missing
2. Your code might expose that bug by passing the PTS set on the input buffer - here you should compare dts & pts to `AV_NOPTS_VALUE` and change to `nil` if it matches (this means it should be most likely a part of the `Common.to_membrane_time_base_truncated`)
3. There is one more bug in the encoder itself - if `pts` is `nil`, `0` is passed to the native part, while the special case in https://github.com/membraneframework/membrane_h264_ffmpeg_plugin/blob/03f7bae79144edb0b0e2895afca2c0552977e3e1/c_src/membrane_h264_ffmpeg_plugin/encoder.c#L163
tests for `AV_NOPTS_VALUE`

They should be fixed before we can make a release of changes in this PR